papas/data/eventplotter.py

```diff
@@ -0,0 +1,143 @@
+        @param detector: detector model used for the simulation
+        @param projections: a list of required projections, eg.
+        @param directory: output directory for images
+        '''
+        self.history = papasevent.history
+        self.papasevent = papasevent  
+        self.helper = HistoryHelper(papasevent)
+        self.detector = detector  
+        if projections is None:
+            projections = ['xy', 'yz']
+        self.projections = projections
+        self.initialized = False 
+        self.directory = directory
+            
+    def plot(self, plottype, screennames, particles_type_and_subtypes, 
+                   clusters_type_and_subtypes, 
+                   num_subgroups=None,
+                   to_file=False):
+        ''' Produces one or more papas event plots depending on options
+        
+        @param plottype: 'event' - plots all elements in an event
+                         'subgroups' - plots are one per subgroup
+        @param screennames: List of names for subscreens (also used to decide whether to plot a single event display 
+                         or a comparison (two plots side by side)) eg ["simulation", "reconstruction"]
+        @param particles_type_and_subtypes: list of particle type_and_subtypes to plot. Length of list must match the length of screennames
+                                      eg ['ps', 'pr'] for simulated particles ('ps') on left, reconstructed particles ('pr') on right
+        @param clusters_type_and_subtypes: list of lists of  clusters to plot, length of list must be same as the length of screennames
+                         eg [['es', 'hs'], ['em', 'hm']] would plot smeared ecals ('es') and smeared hcals ('hs') on left
+                             and merged ecals ('em') and merged hcals ('hm') on right
+        @param num_subgroups: if specified and if plottype = "subgroups" the biggest n subgroups will be plotted
+        @param to_file: if set produces png files of the plots
+        '''
+            
+        #initialise the display (one or two subscreens)
+        self.__init_display(screennames)
+        
+        filename = None
+        if len(particles_type_and_subtypes) != len(screennames) or \
+           len(clusters_type_and_subtypes) != len(screennames)  or \
+           len(screennames) == 0 or len(screennames) > 2:
+            raise(ValueError, "Input arguments are not consistent for event plot")
+        
+        if to_file: #sort out the base of the filename
+            basename = "event_" + str(self.papasevent.iEv)
+            if len(screennames) == 2:
+                basename = "compare_"  + basename          
+        
+        if plottype == "event":  #full event on one plot
+            ids = self.helper.event_ids() 
+            if to_file:
+                filename = basename + ".png"
+            self.plot_ids(ids, screennames, particles_type_and_subtypes, 
+                               clusters_type_and_subtypes, 
```

Comment:
please make sure the indentation is correct before pushing changes (other places with wrong indentation this time). In my case, wing and emacs basically deal with indentation automatically.